### PR TITLE
Fix 12085

### PR DESCRIPTION
**Pull request by dsyme**
Fixes https://github.com/dotnet/fsharp/issues/12085

### CODE REVIEW

**Comment by eiriktsarpalis** [comment]
Thanks! Will this make it to the 6.0 release?

**Comment by dsyme** [comment]
This should also go into release/dev17.0 (if we're not merging main across automatically)

**Comment by dsyme** [comment]
> Thanks! Will this make it to the 6.0 release?

Yes I believe so

File: src/fsharp/IlxGen.fs [hunk 1]
```diff
@@ -107,7 +107,11 @@ let ChooseFreeVarNames takenNames ts =
     ts
 
 /// We can't tailcall to methods taking byrefs. This helper helps search for them
-let IsILTypeByref = function ILType.Byref _ -> true | _ -> false
+let rec IsILTypeByref inp =
+    match inp with
```
Comment (TIHan) [inline src/fsharp/IlxGen.fs:111]: Later we should use `stripILModifiedFromTy` :
```fsharp
match stripILModifiedFromTy inp with
| ILType.Byref _ -> true
| _ -> false
```

**Review by TIHan — APPROVED** [review]

**Comment by dsyme** [comment]
@TIHan We should scan the whole codebase for pattern matchin on ILType

**Pull request merged**